sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/ContainerLockIT.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.data.cosmos;
+
+import com.azure.spring.data.cosmos.core.CosmosTemplate;
+import com.azure.spring.data.cosmos.repository.TestRepositoryConfig;
+import com.azure.spring.data.cosmos.repository.support.CosmosEntityInformation;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = TestRepositoryConfig.class)
+public class ContainerLockIT {
+
+    private static final Duration SHORT_LEASE_DURATION = Duration.ofSeconds(3);
+
+    @Autowired
+    private CosmosTemplate template;
+    private static CosmosTemplate staticTemplate;
+
+    private ContainerLock lock;
+    private ContainerLock otherLock;
+
+    @Before
+    public void setup() {
```

Comment:
As far as I know, the `@Before` and `@After` will be called for each individual test in this class. If that's the case, then we will be creating and deleting containers for each test method. Can you please verify that it runs only once, and we only create and delete container only once per each test class, or per each test suite execution ?

Reply:
That's correct, `@Before` and `@After` will be called for each test method - but setup calls `createContainerIfNotExists` (which only creates the container once) while cleanup calls `releaseLockIgnoreException` (which doesn't delete teh container).  There is a method `afterClassCleanup` which is annotated with `@AfterClass` which only executes once.

Reply:
Makes sense, thanks for the explanation.